### DISCUSSION TITLE
Please provide german voice

### DISCUSSION THREAD

**Discussion by davidak** [coqui-ai/TTS#386] [General]
The quality of the english examples is already very good. What's missing to be useful for me is a model for a german voice.

You can use this issue for related discussions and documenting progress creating it.

**Comment by svenha** [comment]
Has anybody tried the German model listed on https://github.com/coqui-ai/TTS/wiki/Released-Models ?

And @thorstenMueller promised a German model recently: https://github.com/thorstenMueller/deep-learning-german-tts#trained-models Let's see (or hear).

**Comment by erogol** [comment]
If @thorstenMueller shares his models we are more than happy to include in the list and make it available with the ```tts``` end-point and server.  But I guess he uses a different text processing procedure.

**Comment by thorstenMueller** [comment]
I've trained a Tacotron2 DCA model and a compatible WaveGrad and prepare to release this model soon (_might be in march or april_). It uses default (_included_) german phoneme cleaner so there's no additional dependency for text processing. It will be released using default coqui .model.json process and will (_hopefully_) be easy to install using python pip.

**Comment by karussell** [comment]
Thanks for providing this. I'm currently playing with this German voice and it works.

Unfortunately something is wrong because the process is very slow. Instead of seconds (<10s) for a small sentence it takes minutes (>4min)

I used this for example:

```
tts --text "Hallo Welt. Wir können nicht lange reden, da die Zeit knapp ist." --model_name "tts_models/de/thorsten/tacotron2-DCA" --out_path output.wav
```

Can I tweak this somehow? (I also tried with specifying the vocoder_name parameter but this does not change something)

**Reply by erogol** [reply]
it is the vocoder model beong really slow on cpu. If you have cuda use it with --use_cuda

I think @thorstenMueller is also working on making it faster

**Reply by thorstenMueller** [reply]
That's true. I'm currently finetuning a Fullband-MelGan model which has an rtf (real time factor) around 0,5.

Meanwhile you could play with the existing universal vocoder. It generates a metallic voice (that's the reason I'm finetuning it), but works much faster on CPU.

* pip install -U tts
* tts-server --model_name tts_models/de/thorsten/tacotron2-DCA --vocoder_name vocoder_models/universal/libri-tts/fullband-melgan --use_cuda=false

**Reply by karussell** [reply]
Thanks for your fast responses! 

@thorstenMueller after the command you propose I was able to run:

```
tts --text "Hallo Welt. Wir können nicht lange reden, da die Zeit knapp ist." --model_name "tts_models/de/thorsten/tacotron2-DCA" --vocoder_name "vocoder_models/universal/libri-tts/fullband-melgan" --out_path output.wav
```

and it works :) ! (with the small mentioned metallic sound)

Or did you mean that the tts command should somehow connect to this server as it says that it listens at `0.0.0.0:5002` ?

> If you have cuda use it with --use_cuda

Unfortunately I do not have cuda on this machine.

**Reply by karussell** [reply]
I just found out that it stops creating the outout whenever it stumbles over the word "Urteil" and also skips when a verb starts with "ur". See the texts:

 * ich kann über das Urteil noch nicht urteilen weil es einfach nicht geht
 * ich kann über das noch nicht urteilen weil es einfach nicht geht

**Reply by thorstenMueller** [reply]
Hmm, i justed tested your sentence and it worked for me.
Maybe you can test it (_like i did_) with TTS server. Btw. that's the mysterious port 5002.

Try running following command:
```
tts-server --model_name "tts_models/de/thorsten/tacotron2-DCA" --vocoder_name "vocoder_models/universal/libri-tts/fullband-melgan"
```

And open your webbrowser on http://localhost:5002.

**Reply by karussell** [reply]
Ah, nice. Didn't assume there is a simple web UI behind this port! This comes very handy for a toy project I'm planning.

Unfortunately the same problem occurs when I use the texts there. Maybe I use some old version? Is there a way I can print the software and model version?

**Reply by karussell** [reply]
Minor other thing: it does not read the comma if between numbers: "2,7 millionen"

**Reply by thorstenMueller** [reply]
It exists just one released version so there shouldn't be a versioning problem.

For your other question you've to look for a german phoneme cleaner (_preprocessor_) which converts some german abbreviations in a readable form. You'll have similar issues with: "dr., etc., kg., ...".

@repodiac made a nice work here. Maybe take a look at https://github.com/repodiac/german_transliterate

**Reply by repodiac** [reply]
Thanks @thorstenMueller for the referral - yes, if you have questions regarding usage, just let me know @karussell

**Reply by karussell** [reply]
I tried to set it up on a different machine (this time from source and using the server.py) but have the same problems: http://157.90.156.93:5002 if you paste "ich kann über das Urteil noch nicht urteilen weil es einfach nicht geht"
